In [1]:
import scipy.stats as st
import numpy as np

### 1) Известно, что генеральная совокупность распределена нормально со средним квадратическим отклонением, равным 16. Найти доверительный интервал для оценки математического ожидания a с надежностью 0.95, если выборочная средняя M = 80, а объем выборки n = 256.

Расчитаем доверительный интревал, используя Z-критерий: $T_{1,2}=\bar{X}\pm Z_{\frac{a}{2}}* \frac{\sigma}{\sqrt{n}}$

In [2]:
X_bar = 80
a = 1 - 0.95
n = 256
sigma = 16

In [3]:
Z = st.norm.ppf(1 - a / 2)

In [4]:
80 + (Z * sigma / np.sqrt(n))

81.95996398454005

In [5]:
80 - (Z * sigma / np.sqrt(n))

78.04003601545995

**Ответ: Доверительный интервал с точность 0.95 равен  $[78.04; 81.96]$**

### 2) В результате 10 независимых измерений некоторой величины X, выполненных с одинаковой точностью, получены опытные данные: 6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1 Предполагая, что результаты измерений подчинены нормальному закону распределения вероятностей, оценить истинное значение величины X при помощи доверительного интервала, покрывающего это значение с доверительной вероятностью 0,95.

Т-критерий

In [6]:
X = np.array([6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1])

In [7]:
n = len(X)
sigma = X.std()
mean = X.mean()

In [8]:
mean

6.590000000000001

In [9]:
# По таблице найдем значение t
t = 2.262

In [10]:
T1 = mean - (t * sigma / np.sqrt(n))
T1

6.28408592912388

In [11]:
T2 = mean + (t * sigma / np.sqrt(n))
T2

6.895914070876121

**Ответ: интервал равен  $[6.28; 6.9]$. С доверительной вероятностью 95% величина $X=6.59$**

### 3) Утверждается, что шарики для подшипников, изготовленные автоматическим станком, имеют средний диаметр 17 мм. Используя односторонний критерий с α=0,05, проверить эту гипотезу, если в выборке из n=100 шариков средний диаметр оказался равным 17.5 мм, а дисперсия известна и равна 4 мм.

Сформулируем основную и альтернативную гипотезы.

In [12]:
mu_0 = 17
mu = 17.5
a = 0.05
n = 100
variance = 4
std = np.sqrt(variance)

$H_{0}:$ Диаметр = 17 мм, $\mu=\mu_{0}$

$H_{1}:$ Диаметр = 17.5 мм, $\mu>\mu_{0}$

Выберем статистический критерий, по которому будем проводить проверку.

Z-критерий, так как известна $\mu$ генеральной совокупности ($\mu_{0}$)

Зададим уровень значимости  𝛼 , или вероятность, с которой будем выполнять данную гипотезу.

In [13]:
a = 0.05

Определим границы области гипотезы.

$Z_{observable}=\frac{\bar{X}-\mu_{0}}{\frac{\sigma}{\sqrt{n}}}$

In [14]:
z = st.norm.ppf(1 - a)
z

1.6448536269514722

In [15]:
Z_obs = (mu - mu_0) / (std / np.sqrt(n))
Z_obs

2.5

Подведем итоги и сформулируем вывод.

**Ответ: При уровне значимости 5 %, отвергаем гипотезу $H_{0}$, так как $Z_{observable} > z$**

### 4) Продавец утверждает, что средний вес пачки печенья составляет 200 г. Из партии извлечена выборка из 10 пачек. Вес каждой пачки составляет: 202, 203, 199, 197, 195, 201, 200, 204, 194, 190. Известно, что их веса распределены нормально. Верно ли утверждение продавца, если учитывать, что доверительная вероятность равна 99%?

In [16]:
X = np.array([202, 203, 199, 197, 195, 201, 200, 204, 194, 190])

In [17]:
sigma = X.std()
mean = X.mean()
n = len(X) # 10

$H_{0}:$ Вес = 200 г, $\mu=\mu_{0}$

$H_{1}:$ Вес $\neq$ 200 г, $\mu\neq\mu_{0}$

Выберем статистический критерий, по которому будем проводить проверку.

T-критерий, так как неизвестна $\mu$ генеральной совокупности ($\mu_{0}$)

In [18]:
# По таблице найдем значение t
t = 3.250

In [19]:
T1 = mean - (t * sigma / np.sqrt(n))
T1

194.1578735048366

In [20]:
T2 = mean + (t * sigma / np.sqrt(n))
T2

202.8421264951634

Подведем итоги и сформулируем вывод.

**Ответ: С доверительной вероятностью 99%, утверждение продавца верно - средний вес пачки 200 г.**